### PR TITLE
[Mellanox] add FAN led management in thermal control daemon

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
What I did:
1. Set led to red and trigger syslog when fan is broken
2. Add fan drawer information to redis database
3. Trigger a syslog when any fan is absence or broken to alert user that "Insufficient fan number"

### CODE REVIEW

File: sonic-thermalctld/scripts/thermalctld [hunk 1]
```diff
@@ -64,27 +64,61 @@ def log_on_status_changed(normal_status, normal_log, abnormal_log):
...
         self.under_speed = False
         self.over_speed = False
         self.invalid_direction = False
 
+    @classmethod
+    def get_bad_fan_count(cls):
+        return cls.absence_fan_count + cls.fault_fan_count
+
+    @classmethod
+    def reset_bad_fan_count(cls):
```
Comment (liat-grozovik) [inline sonic-thermalctld/scripts/thermalctld]: why not having this function as generic. can be called even if not bad fan. just reset counters

File: sonic-thermalctld/scripts/thermalctld [hunk 2]
```diff
@@ -142,7 +176,7 @@ class FanStatus(object):
         Indicate the Fan works as expect
         :return: True if Fan works normal else False
         """
-        return self.presence and not self.under_speed and not self.over_speed and not self.invalid_direction
+        return self.presence and self.status and not self.under_speed and not self.over_speed and not self.invalid_direction
```
Comment (liat-grozovik) [inline sonic-thermalctld/scripts/thermalctld]: it becomes to complicated line. can you make it more readable?

File: sonic-thermalctld/scripts/thermalctld [hunk 3]
```diff
@@ -176,33 +210,51 @@ class FanUpdater(object):
...
+                except Exception as e:
+                    logger.log_warning('Failed to update FAN status - {}'.format(e))
+                fan_index += 1
 
         for psu_index, psu in enumerate(self.chassis.get_all_psus()):
             psu_name = try_get(psu.get_name, 'PSU {}'.format(psu_index))
             for fan_index, fan in enumerate(psu.get_all_fans()):
                 try:
-                    self._refresh_fan_status(fan, fan_index, '{} FAN'.format(psu_name))
+                    # PSU can be treated as a fan drawer
```
Comment (liat-grozovik) [inline sonic-thermalctld/scripts/thermalctld]: not sure i understand the logic that PSU is treated as FAN drawer. why?

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

File: sonic-thermalctld/scripts/thermalctld (hunk 3 above)
Comment (Junchao-Mellanox) [reply sonic-thermalctld/scripts/thermalctld]: Will pass a None value here.

**Review by keboliu — APPROVED** [review]

**Comment by Junchao-Mellanox** [comment]
Community PR https://github.com/Azure/sonic-platform-daemons/pull/54

**Pull request closed without merging**